### ISSUE TITLE
eraser and recover brush for fabricjs( canvas)

### ISSUE THREAD

**Issue by komagic** [fabricjs/fabric.js#5187]
I created some methods to implement eraser and recover brush in fabricjs（by globalCompositeOperation）,but i still dont know  how to use original canvas methods
 in fabricjs?

**Comment by asturur** [comment]
Hi @komagic, i do not understand your question.

Can you rephrase it, please?

**Comment by komagic** [comment]
how to create a path with alpha feather edges( fabricjs)?
![2018-08-22 3 30 59](https://user-images.githubusercontent.com/13415767/44449386-92642100-a620-11e8-92ae-b21274c7041b.png)

**Comment by asturur** [comment]
There is no such a tool.
If you can define a path for your shape you can overlay a series of strokes with different opacity to imitate the effect.

**Comment by komagic** [comment]
how to copy the content of the canvas to another canvas correctly?   use todataURL pls. (do not use toJson)

**Comment by komagic** [comment]
success! Then I want to drop the picture back to the old fabric canvas, the picture should be put correctly?

**Comment by asturur** [comment]
i can t really follow step by step your project. You should make a good open stack overflow snippet and ask help on stackoverflow. Here we can t provide this kind of help

**Comment by Euphoria9702** [comment]
我最近也在想用fabric.js您当时实现的内容，demo可以放到git上给我参考学习一下吗，哈哈

**Comment by Euphoria9702** [comment]
@ @komagic

**Comment by ShaMan123** [comment]
@asturur can shadow be used here?

**Comment by ShaMan123** [comment]
> haha，i am try to create a simple editor by fabricjs， i created an eraser and a recovery brush, as show below:
> ![2018-08-22 10 09 23](https://user-images.githubusercontent.com/13415767/44448736-adce2c80-a61e-11e8-81ba-4b0fbc2df2ce.png)
> 
> however, i do not know how to create a path with feather edges by fabricjs，can you show me some samples?

This is a cool feature

**Comment by ShaMan123** [comment]
Tried it out, looks interesting:
https://codesandbox.io/s/exciting-architecture-bcgej?file=/src/App.tsx

hit the `erase` button

**Comment by Euphoria9702** [comment]
how to created  recovery brush @komagic

**Comment by ShaMan123** [comment]
Try finddling with `inverted erasing`
https://codesandbox.io/s/blissful-sutherland-yk05j?file=/src/EraserBrush.js:25469-25473
It simply changes the `globalCompositeOperation `
```js
(path | ctx).globalCompositeOperation = this.inverted
          ? "source-over"
          : "destination-out";

```
The problem is the rendering of the brush in such a case.
It will draw color and I would expect it won't and only recover erased paths.
This might be possible but will require some work.
Consider this:
https://github.com/fabricjs/fabric.js/blob/master/src/mixins/eraser_brush.mixin.js#L420
When erasing occurs, the path (generated by mouse move) clips the top ctx and reveals the bottom ctx.
   * This achieves the desired effect of seeming to erase only erasable objects.
   * After erasing is done the created path is added to all intersected objects' `clipPath` property.

To achieve an inverted/recovery effect the canvas should be prepared differently:
1. Render all objects on bottom ctx without their erasers
2. Render some part of the eraser clip path
3. set `globalCompositionOpertion =  "source-over"`
4. On mouse up add the path as any eraser path (with existing logic)

**Comment by Euphoria9702** [comment]
It will draw color and I would expect it won't and only recover erased paths.

**Comment by ShaMan123** [comment]
Indeed, as I stated in the PR

**Comment by ShaMan123** [comment]
solved

**Comment by ShaMan123** [comment]
The eraser has moved https://github.com/ShaMan123/erase2d
